lib/logstash/inputs/sqs.rb

```diff
@@ -151,28 +151,39 @@ def run(output_queue)
   end
 
   private
+
   # Runs an AWS request inside a Ruby block with an exponential backoff in case
   # we experience a ServiceError.
   #
-  # @param [Integer] max_time maximum amount of time to sleep before giving up.
-  # @param [Integer] sleep_time the initial amount of time to sleep before retrying.
   # @param [Block] block Ruby code block to execute.
-  def run_with_backoff(max_time = MAX_TIME_BEFORE_GIVING_UP, sleep_time = BACKOFF_SLEEP_TIME, &block)
-    next_sleep = sleep_time
-
+  def run_with_backoff(&block)
+    sleep_time = BACKOFF_SLEEP_TIME
     begin
       block.call
-      next_sleep = sleep_time
-    rescue Aws::SQS::Errors::ServiceError => e
-      @logger.warn("Aws::SQS::Errors::ServiceError ... retrying SQS request with exponential backoff", :queue => @queue, :sleep_time => sleep_time, :error => e)
-      sleep(next_sleep)
-      next_sleep =  next_sleep > max_time ? sleep_time : sleep_time * BACKOFF_FACTOR 
-
+      sleep_time = BACKOFF_SLEEP_TIME
```

Comment:
Do we need to reset this here?

Reply:
:1st_place_medal: we do not (wasn't obvious), left it after copy-pasting the old code.